src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.threadpool.ThreadPool;
+
+/**
+ * An abstract class that provides sample methods required by extensions
+ */
+public abstract class BaseExtension implements Extension {
+    /**
+     * A client to make requests to the system
+     */
+    protected SDKClient client;
+
+    /**
+     * A service to allow watching and updating cluster state
+     */
+    protected ClusterService clusterService;
+
+    /**
+     * A service to allow retrieving an executor to run an async action
+     */
+    protected ThreadPool threadPool;
```

Comment:
Do we need a threadPool in the SDK?

Reply:
We need it in the BaseExtension class for createComponents, and this way the same threadpool can be stored for each extension

Reply:
I had an offline discussion with @dbwiddis top keep threadpool either in SDK or create a new in extension. For now we can keep it in SDK but we might want to revisit this and discuss for the right place.

Reply:
Sure

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/183